app/addons/documents/sidebar/sidebar.react.jsx

```diff
@@ -110,38 +120,92 @@ function (app, FauxtonAPI, React, ReactDOM, Stores, Actions,
       );
     }
-
   });
 
+
   var IndexSection = React.createClass({
 
     propTypes: {
       urlNamespace: React.PropTypes.string.isRequired,
-      databaseName: React.PropTypes.string.isRequired,
+      indexLabel: React.PropTypes.string.isRequired,
+      database: React.PropTypes.object.isRequired,
       designDocName: React.PropTypes.string.isRequired,
       items: React.PropTypes.array.isRequired,
       isExpanded: React.PropTypes.bool.isRequired,
-      selectedIndex: React.PropTypes.string.isRequired
+      selectedIndex: React.PropTypes.string.isRequired,
+      onDelete: React.PropTypes.func.isRequired,
+      onClone: React.PropTypes.func.isRequired
     },
 
     createItems: function () {
-      return _.map(this.props.items, function (index, key) {
-        var href = FauxtonAPI.urls(this.props.urlNamespace, 'app', this.props.databaseName, this.props.designDocName);
-        var className = (this.props.selectedIndex === index) ? 'active' : '';
+
+      // sort the indexes alphabetically
+      var sortedItems = this.props.items.sort();
+
+      return _.map(sortedItems, function (indexName, index) {
+        var href = FauxtonAPI.urls(this.props.urlNamespace, 'app', this.props.database.id, this.props.designDocName);
+        var className = (this.props.selectedIndex === indexName) ? 'active' : '';
 
         return (
-          <li className={className} key={key}>
+          <li className={className} key={index}>
             <a
-              id={this.props.designDocName + '_' + index}
-              href={"#/" + href + index}
+              id={this.props.designDocName + '_' + indexName}
+              href={"#/" + href + indexName}
               className="toggle-view">
-              {index}
+              {indexName}
             </a>
+
+            <OverlayTrigger
+              ref="indexMenu"
+              trigger="click"
+              placement="bottom"
+              rootClose={true}
+              overlay={
+                <Popover id="index-menu-component-popover">
+                  <ul>
+                    <li onClick={this.indexAction.bind(this, 'edit', { indexName: indexName, onEdit: this.props.onEdit })}>
+                      <span className="fonticon fonticon-file-code-o"></span>
+                      Edit
+                    </li>
+                    <li onClick={this.indexAction.bind(this, 'clone', { indexName: indexName, onClone: this.props.onClone })}>
+                      <span className="fonticon fonticon-files-o"></span>
+                      Clone
+                    </li>
+                    <li onClick={this.indexAction.bind(this, 'delete', { indexName: indexName, onDelete: this.props.onDelete })}>
+                      <span className="fonticon fonticon-trash"></span>
+                      Delete
+                    </li>
+                  </ul>
+                </Popover>
+              }>
+              <span className="index-menu-toggle fonticon fonticon-wrench2"></span>
```

Comment:
can we make the hover-area a bit larger -- i am quite a web person but i have problems to hit the wrench icon in the sidebar from time to time